evalml/automl/automl_algorithm/iterative_algorithm.py

```diff
@@ -37,7 +48,22 @@ def __init__(self,
             ensembling (boolean): If True, runs ensembling in a separate batch after every allowed pipeline class has been iterated over. Defaults to False.
             pipeline_params (dict or None): Pipeline-level parameters that should be passed to the proposed pipelines.
             _frozen_pipeline_parameters (dict or None): Pipeline-level parameters are frozen and used in the proposed pipelines.
+            _estimator_family_order (list(ModelFamily) or None): specify the sort order for the first batch. Defaults to _ESTIMATOR_FAMILY_ORDER.
         """
+        self._estimator_family_order = _estimator_family_order or _ESTIMATOR_FAMILY_ORDER
+        indices = []
+        pipelines_to_sort = []
+        pipelines_end = []
+        for pipeline in allowed_pipelines or []:
+            if pipeline.model_family in self._estimator_family_order:
+                indices.append(self._estimator_family_order.index(pipeline.model_family))
+                pipelines_to_sort.append(pipeline)
+            else:
+                pipelines_end.append(pipeline)
+        pipelines_start = [pipeline for _, pipeline in (sorted(zip(indices, pipelines_to_sort),
+                                                               key=lambda pair: pair[0]) or [])]
+        allowed_pipelines = pipelines_start + pipelines_end
```

Comment:
There were a few options for where to put this code:
1. In `IterativeAlgorithm.__init__`, when `allowed_pipelines` is passed in from `AutoMLAlgorithm.__init__`
2. In `AutoMLAlgorithm.__init__`, when `allowed_pipelines` is first created.
3. In `get_estimators` and the underlying utils, which define the order for the `allowed_pipelines` list created in `AutoMLAlgorithm.__init__`

I went with option 1.

My reasoning: the sort order of the first batch, and in fact the definition of what pipelines will be included in what automl batch, is owned by the automl algorithm instance. And `IterativeAlgorithm` defines our current default automl algorithm.